addons/sale_coupon/models/sale_order.py

```diff
@@ -45,6 +45,7 @@ def copy(self, default=None):
     def action_confirm(self):
         self.generated_coupon_ids.write({'state': 'new'})
         self.applied_coupon_ids.write({'state': 'used'})
+        self.recompute_coupon_lines()
```

Comment:
why is this necessary?

Reply:
because the coupon is not automatically updated each time we modify the quantity (linked to your message on the previous PR) suddenly I force the recalculation of the coupon each time the quotation is confirmed to have the right numbers

it is not the case in 12.0, only in 13.0 and 14.0

Reply:
Even if this seems odd it is the current behavior of the module now ;)
https://github.com/odoo/enterprise/commit/5f6371ee6ade2a29dac8acccf60dc8d65d2afec1